app/models/pawn.rb

```diff
@@ -19,4 +19,15 @@ def first_move?
     ((position_y).eql?(1) && white? || (position_y).eql?(6) && black?)
   end
   
+  def promotable?
+    (color.eql?('black') && position_y.eql?(0) || color.eql?('white') && position_y.eql?(7))
+  end
+  
+  def promote!(x, y)
+    if promotable?
+      pawn_piece = present_piece(x, y)
+      pawn_piece.update_attributes(type: 'Queen', special: 'promoted')
+    end
+  end
+  
```

Comment:
This is the most obvious and easiest way to do this so I am all for it.... but technically you get a choice as to what you want to promote to. But if everyone else is ok with this I am all for it

Reply:
Yeah I agree, I figured once everything is up and running we can create a pop up in the UI to let user select which one. But for now since queen is most likely what you would choose we can rock with this.